### PR TITLE
Update neovim build and libvterm

### PR DESCRIPTION
**Pull request by hhoeflin**
Updating the neovim recipe to:
- Add version 0.4.4
- Fix build errors for github branches master and stable
  - Lua version was set at 5.1:5.2, but neovim only allows Lua version 5.1, not 5.2
  - Dependency versions were picked incorrectly for labels stable and master
  - For centos7, a newer version of git was required. Added as a build dependency
  - For version 0.2.0, building failed due to issues with dependency libvterm

Updating libvterm:
- Version 0.0.0 was pointing to the incorrect file (typo?)
- Version 0.1.4 was added

### CODE REVIEW

**Comment by hhoeflin** [comment]
Sorry - is my first PR here. Any next steps I need to do?

**Comment by hhoeflin** [comment]
Anything missing?

File: var/spack/repos/builtin/packages/libvterm/package.py
Comment (healther) [inline var/spack/repos/builtin/packages/libvterm/package.py:6]: drive by? Not sure if this is a problem from our style guides (I would assume not given that nothing failed)?

File: var/spack/repos/builtin/packages/neovim/package.py
Comment (healther) [inline var/spack/repos/builtin/packages/neovim/package.py:42]: why the additional restriction?

**Review by healther — COMMENTED** [review]
In principle you'll just need to wait for one of the maintainers to approve and merge. I made a couple of comments, especially the added restrictions deserves a reason (in the commit message)

Also I seem to remember that `@master` should satisfy `@0.4:` but that doesn't seem to be the case (according to your commit message)

File: var/spack/repos/builtin/packages/libvterm/package.py
Comment (hhoeflin) [reply var/spack/repos/builtin/packages/libvterm/package.py:6]: what does "drive by" mean?

File: var/spack/repos/builtin/packages/neovim/package.py
Comment (hhoeflin) [reply var/spack/repos/builtin/packages/neovim/package.py:42]: which restriction exactly?

Comment (hhoeflin) [reply var/spack/repos/builtin/packages/neovim/package.py:42]: which restriction exactly?

Comment (healther) [reply var/spack/repos/builtin/packages/neovim/package.py:42]: You switched from "everything up to 0.2.0" to "exactly 0.2.0"

File: var/spack/repos/builtin/packages/libvterm/package.py
Comment (healther) [reply var/spack/repos/builtin/packages/libvterm/package.py:6]: non-intended change. I'm not sure if we have style guides that say "there should be exactly one empty line between the license identifier and the first statement of the file.

Comment (hhoeflin) [reply var/spack/repos/builtin/packages/libvterm/package.py:6]: Sure - happy to remove that line ... I did not notice.

File: var/spack/repos/builtin/packages/neovim/package.py
Comment (hhoeflin) [reply var/spack/repos/builtin/packages/neovim/package.py:42]: Yes - I tried all lower version than 0.2.0 and they all don't compile - and none of the are mentioned by hash in the file. Happy to include for everything lower.

Comment (healther) [reply var/spack/repos/builtin/packages/neovim/package.py:42]: ah, if you know that the older versions do not build then it's good to remove it. In general we try to keep the dependencies as general as possible. I.e. while the package does not contain information about older versions right now, adding new ones can happen anytime, and we try to minimise the overhead and let the concretiser do as much work as possible. 

In this case I'd mention the additional restriction in the commit message, including that you tried older versions and they didn't work. (You could also add the older versions)

Comment (hhoeflin) [reply var/spack/repos/builtin/packages/neovim/package.py:42]: ok - will add it in the commit message. And I agree that while generally I am all in favor of fixing also for older versions ... in this case I will abstain as this is not likely to be a dependency of something else as it is an editor ...

File: var/spack/repos/builtin/packages/neovim/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/neovim/package.py]: Is this only needed to download the package? In that case I wouldn't bother making this an explicit dependency

**Review by adamjstewart — COMMENTED** [review]
`@master` should satisfy `@1:`, but `@stable` won't.

**Comment by hhoeflin** [comment]
Removed the git dependency. Anything else?

**Comment by adamjstewart** [comment]
You should be able to remove `,master` from all of these `when=` args.

**Comment by hhoeflin** [comment]
Done.

**Review by adamjstewart — APPROVED** [review]
@healther does this look good to you now?

**Review by healther — APPROVED** [review]

**Comment by healther** [comment]
Yep

**Pull request merged**
